src/gpu_allocator.zig

```diff
@@ -0,0 +1,623 @@
+const std = @import("std");
+
+pub const Error = error{
+    OutOfMemory,
+    FreeingZeroSizeAllocation,
+    InvalidAllocation,
+    NoCompatibleMemoryFound,
+    Other,
+};
+
+pub const MemoryLocation = enum {
```

Comment:
I'd like for this allocator to be generic enough for other purposes (e.g. D3D12 descriptors, sub-allocating CPU memory).  It doesn't look like it's used here?